Item(by='int_19h', descendants=None, kids=[24658054], score=None, time=1601590171, title=None, item_type='comment', url=None, parent=24655478, text='Performing an operation that can fail in a constructor is not a problem, because then the initializer throws, and you don&#x27;t get an object at all - so there&#x27;s nothing for you to check if it&#x27;s in a valid state or not.<p>On the other hand, this is exactly the problem that arises if you <i>refuse</i> to use exceptions, because then the constructor has no way to signal that it failed. Since constructors, by their very nature (acquiring resources, validating initializers etc), <i>can</i> fail, you have to resort to &quot;is it valid?&quot; flags, and your API clients have to check those flags or risk unspecified behavior.')